lib/entities/parse_address_from_domain.dart

```diff
@@ -96,6 +100,14 @@ class AddressResolver {
         return ParsedAddress.fetchUnstoppableDomainAddress(address: address, name: text);
       }
 
+      if (text.endsWith(".eth")) {
+        var wallet = getIt.get<AppStore>().wallet!;
+        final address = await EnsRecord.fetchEnsAddress(text, wallet: wallet);
```

Comment:
also you don't need to pass the whole wallet, you only need the wallet type which is already existing here, so you can remove this and just pass the already existing wallet type

Reply:
I pass the wallet because that way we can use whatever the current ethereum node is if we are on an ethereum wallet, rather than just rely on the one that's hardcoded in ens_record.dart

Reply:
ok, but let's not use a dependency like this, injecting it would be better for maintainability, please check the last commit